### PR TITLE
Add testsets

### PR DESCRIPTION
**Pull request by amontoison**
@dpo 
I removed all outputs in the tests and each method is tested within a `testset`.

### CODE REVIEW

File: test/check_min_norm.jl [hunk 1]
```diff
@@ -10,6 +10,5 @@ function check_min_norm(A, b, x; λ=0.0)
   QR = qr(AI')
   xmin = QR.Q * (QR.R' \ b)
   xmin_norm = norm(xmin)
-  @printf("‖x - xmin‖ = %7.1e\n", norm(xI - xmin))
```
Comment (dpo) [inline test/check_min_norm.jl:13]: We should test that they’re close.

File: test/test_aux.jl [hunk 2]
```diff
@@ -97,4 +97,6 @@ function test_aux()
   Krylov.kones(Vector{Float64}, 10) == ones(10)
 end
 
-test_aux()
+@testset "aux" begin
+  test_aux()
```
Comment (dpo) [inline test/test_aux.jl:101]: Do we still need a function here? Why not paste the contents of `test_aux()` inside the testset?

Comment (amontoison) [reply test/test_aux.jl:101]: For the allocation tests we need a function but in that case it's not required. The advantage of using functions is that all variables are in a local scope.

File: test/check_min_norm.jl (hunk 1 above)
Comment (amontoison) [reply test/check_min_norm.jl:13]: We already test that they're close : 
https://github.com/JuliaSmoothOptimizers/Krylov.jl/blob/master/test/test_lnlq.jl#L32
https://github.com/JuliaSmoothOptimizers/Krylov.jl/blob/master/test/test_craig.jl#L23
https://github.com/JuliaSmoothOptimizers/Krylov.jl/blob/master/test/test_cgne.jl#L23

I only removed the `@printf` message.

File: test/test_aux.jl (hunk 2 above)
Comment (dpo) [reply test/test_aux.jl:101]: Same with testsets I think since they are between `begin .. end`.

Comment (amontoison) [reply test/test_aux.jl]: I updated the PR, I only kept a function for allocation tests. The type of `MinresSolver` is not correcly infered within a `begin ... end`.

**Comment by dpo** [comment]
OMG! So much better!!! Thanks!

I think we lost a little coverage because we’re no longer exercising the `show` functions. Abel has a way of testing those with `IOBuffer`s. I’ll open an issue for later.

**Pull request merged**